app/src/main/java/net/gsantner/markor/activity/MainActivity.java

```diff
@@ -104,45 +105,22 @@ public void onPageSelected(int position) {
 
         // Switch to tab if specific folder _not_ requested, and not recreating from saved instance
         final int startTab = _appSettings.getAppStartupTab();
-        if (startTab != R.id.nav_notebook && savedInstanceState == null && getIntentDir(getIntent(), null) == null) {
+        if (startTab != R.id.nav_notebook && savedInstanceState == null && GsFileUtils.getValidIntentDir(getIntent(), null) == null) {
             _viewPager.postDelayed(() -> _viewPager.setCurrentItem(tabIdToPos(startTab)), 100);
         }
     }
 
     @Override
     protected void onNewIntent(final Intent intent) {
         super.onNewIntent(intent);
-        final File dir = getIntentDir(intent, null);
+        final File dir = GsFileUtils.getValidIntentDir(intent, null);
         final GsFileBrowserFragment frag = getNotebook();
         if (frag != null && dir != null) {
             frag.post(() -> frag.getAdapter().setCurrentFolder(dir));
             _bottomNav.postDelayed(() -> _bottomNav.setSelectedItemId(R.id.nav_notebook), 10);
         }
     }
 
-    private static File getIntentDir(final Intent intent, final File fallback) {
```

Comment:
Moved to GsFileUtils

Reply:
please have that one in ./app/src/main/java/net/gsantner/markor/util/MarkorContextUtils.java